velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -225,5 +225,28 @@ TEST_F(DateTimeFunctionsTest, lastDay) {
   EXPECT_EQ(lastDayFunc(std::nullopt), std::nullopt);
 }
 
+TEST_F(DateTimeFunctionsTest, dateAdd) {
+  const auto dateAdd = [&](std::optional<int32_t> date,
+                           std::optional<int32_t> value) {
+    return evaluateOnce<int32_t, int32_t>(
+        "date_add(c0, c1)", {date, value}, {DATE(), INTEGER()});
+  };
+
+  // Check null behaviors
+  EXPECT_EQ(std::nullopt, dateAdd(std::nullopt, 1));
```

Comment:
I think you need to add some additional code to handle the case when the second argument is null (and the function should throw). It probably means that you need to implement `callNullable()` instead of `call()`.

Reply:
Spark returns null if either argument is null. Sorry for the mistake. Corrected in PR description. https://github.com/apache/spark/blob/7d96f12784de7f49eaf2bb0a0d8b5cb17d2ecf63/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala#L409-L413